Sources/NIO/AddressedEnvelope.swift

```diff
@@ -20,15 +20,35 @@
 public struct AddressedEnvelope<DataType> {
     public var remoteAddress: SocketAddress
     public var data: DataType
+    /// Any MetaData associated with this `AddressedEnvelope`
+    public var metaData : MetaData? = nil
 
     public init(remoteAddress: SocketAddress, data: DataType) {
         self.remoteAddress = remoteAddress
         self.data = data
     }
+    
+    /// Any MetaData associated with an AddressedEnvelope
+    public struct MetaData {
+        /// Details of any congestion state.
+        public var ecnState : NIOEcnState
+    }
 }
 
 extension AddressedEnvelope: CustomStringConvertible {
     public var description: String {
         return "AddressedEnvelope { remoteAddress: \(self.remoteAddress), data: \(self.data) }"
     }
 }
+
+/// Possible ECN States
+public enum NIOEcnState {
```

Comment:
As "ECN" is an acronym this should read `NIOECNState`.

Reply:
Can this be conformed to `Hashable`?

Reply:
done.